presto-main/src/main/java/com/facebook/presto/sql/MaterializedViewUtils.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql;
+
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.sql.analyzer.Analysis;
+import com.facebook.presto.sql.analyzer.MaterializedViewPlanValidator;
+import com.facebook.presto.sql.tree.Query;
+import com.facebook.presto.sql.tree.Statement;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.ConnectorMaterializedViewDefinition.TableColumn;
+import static com.facebook.presto.sql.analyzer.MaterializedViewPlanValidator.MaterializedViewPlanValidatorContext;
+
+public class MaterializedViewUtils
+{
+    private MaterializedViewUtils() {}
+
+    public static void validateMaterialziedViewQueryPlan(Statement query)
+    {
+        MaterializedViewPlanValidator validator = new MaterializedViewPlanValidator(query);
+        validator.process(query, new MaterializedViewPlanValidatorContext());
+    }
+
+    public static Map<String, Map<SchemaTableName, String>> computeMaterializedViewToBaseTableColumnMappings(Query viewQuery, Analysis analysis)
+    {
+        ImmutableMap.Builder<String, Map<SchemaTableName, String>> fullColumnMapping = ImmutableMap.builder();
+
+        Map<String, Map<SchemaTableName, String>> originalColumnMapping = analysis.getOriginalColumnMapping(viewQuery);
```

Comment:
Should we pull getOriginalColumnMapping as well in this class?

Reply:
A good suggestion